django/db/models/query.py

```diff
@@ -2416,7 +2416,7 @@ def prefetch_related_objects(model_instances, *related_lookups):
                         new_obj = list(obj._prefetched_objects_cache.get(through_attr))
                     else:
                         try:
-                            new_obj = getattr(obj, through_attr)
+                            new_obj = getattr(obj, to_attr)
```

Comment:
This fixes the test below and our failing tests. It also doesn't break any django test as far as I could tell.

It makes sense that we read the `to_attr` on `obj` and not the `through_attr` as, if specified, the former one will be set on the instance. Using `through_attr` when a `to_attr` was given will cause a cache miss and trigger a new, undesired query.